CHANGES.rst

```diff
@@ -1,10 +1,17 @@
 Changelog
 =========
 
-2.2.1 (unreleased)
+2.3.0 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Add ``import_path`` to ``upgradeDepends`` directive.
+  This avoids the need to register a full profile, when you only need
+  to import a few files once in an upgrade step.
+  The path must be within the path of a package.
+  By default this is the path of the profile for which this is an upgrade.
+  You can also specify a path in a different package: ``other.package:profile/path``.
+  You can combine this with ``import_steps``, but not with ``import_profile``.
+  Note that ``metadata.xml`` is never read, so you cannot use dependencies.
```

Comment:
> `GenericSetup` has 2 kinds of dependencies: "step" and "profile" dependencies. To be specific, I suggest `(profile) dependencies` in place of `dependencies`.

That is indeed clearer. Fixed.

> Why do you treat `metadata.xml` specially? Would it not be more natural to view the new feature as just a different way to refer to a profile including the treatment of `metadata.xml` (if it exists).

The same is true for the already existing import from tar archives. A `metadata.xml` in there is ignored. This file is only read when [registering a profile](https://github.com/zopefoundation/Products.GenericSetup/blob/master/src/Products/GenericSetup/registry.py#L759).

I sometimes wished this was different: when you edit `metadata.xml` of a standard profile, you have to restart your instance, otherwise it has no effect. That could be something for a different PR.

Here the idea is: if you want metadata, then you should use a standard profile. 
I would use this new feature to ease upgrading. Instead of a `metadata.xml`, I would just add an `upgradeDepends` for each other profile that I need. The other two items you could add in `metadata.xml` are version and description, both of which we don't need.

> Profile ids have the form `profile-`_package_`:`_name_. This suggests a path alternative of the form `path-`_package_`:`_path_. It would allow to universally specify profiles via paths: no need to add an additional parameter to several methods, no need to treat `metadata.xml` specially, True, the parameter `profile_id` would with this extension better have the name `profile_spec`, but I think a documentation note of the form: a "profile id" has either the form `profile-...` or `path-...` would be sufficient to explain the a profile can be identified either via a registered profile name or a file system path.

That could be an option. I could try that.